### DISCUSSION TITLE
[Feature request] add auto start option

### DISCUSSION THREAD

**Discussion by RecursiveFunctions** [vicwomg/pikaraoke#292] [General]
Not necessarily a bug since this is self inflicted and it's quite possible I fat-fingered something while following the instructions. When I reboot the pi after setting it to autostart pikaraoke, it hangs on the 'Welcome' screen when booting up. 

I'm using raspbery pi OS. Debian 12 (bookworm)

Either way, it got me thinking: why leave this to chance for a user to mess it up like that? I think adding an option for it in setup.sh would be good.

I could write it up myself and do a PR.  

Would this need to be written cross-platform a la python? Or would a BASH script suffice? I only had the pi in mind for this idea, but I could look up how to do it on MacOS and windows too.

**Comment by vicwomg** [comment]
Hmm, I just tried it myself on a fresh pi, the instructions worked fine. Do you know what your mistake was? I'm finding it hard to imagine what kind of autostart would prevent loading. If anything, it would just error out and continue.

I don't think it's necessary to add scripted support for this, we can probably just add a disclaimer to the instructions to back up your data and make sure you can ssh in remotely to fix any errors that might occur. But I'd still be curious how one could get into this state.

**Comment by RecursiveFunctions** [comment]
I'ma backup a working image and retry this to see if i can reproduce it

**Comment by RecursiveFunctions** [comment]
Naw I definitely fat-fingered it the first time.

It may not be necessary but I think every little bit towards improving the user experience is worthwhile.

Would you be opposed to adding this as an option?

**Comment by TekyTech** [comment]
I’d like to wholeheartedly support this proposal. 
More over, I’d like to be able to turn auto start on or off any time, from the pikaraoke interface. 
I’ve had situations when it was a major hassle without it. I can explain more reasons if needed, otherwise I won’t write what might be boring or unnecessary for others.
